src/main/java/com/wanderersoftherift/wotr/rift/RiftDifficultyEvents.java

```diff
@@ -0,0 +1,57 @@
+package com.wanderersoftherift.wotr.rift;
+
+import com.wanderersoftherift.wotr.WanderersOfTheRift;
+import com.wanderersoftherift.wotr.core.rift.RiftData;
+import net.minecraft.core.Holder;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.Mob;
+import net.minecraft.world.entity.ai.attributes.Attribute;
+import net.minecraft.world.entity.ai.attributes.Attributes;
+import net.minecraft.world.entity.player.Player;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.neoforge.event.entity.EntityJoinLevelEvent;
+
+@EventBusSubscriber(modid = WanderersOfTheRift.MODID, bus = EventBusSubscriber.Bus.GAME)
+public class RiftDifficultyEvents {
+
+    @SubscribeEvent
+    public static void onEntitySpawned(EntityJoinLevelEvent event) {
+        if (event.getLevel().isClientSide() || event.getEntity() instanceof Player) {
+            return;
+        }
+        if (event.getLevel() instanceof ServerLevel serverLevel && RiftData.isRift(serverLevel)
+                && event.getEntity() instanceof LivingEntity livingEntity) {
+            applyDifficultyToEntity(livingEntity, serverLevel);
+        }
+    }
+
+    private static void applyDifficultyToEntity(LivingEntity livingEntity, ServerLevel serverLevel) {
+        int tier = RiftData.get(serverLevel).getTier();
+        if (livingEntity instanceof Mob mob) {
```

Comment:
If it is only going to work for mobs, should the mob check be moved into `onEntitySpawned`?

Reply:
This should eventually change into a more detailed system that detemrines a difference between player summoned and rift spawned. I kept it to mob to prevent oddities before we have something more complex in place.